bedrock/firefox/tests/test_views.py

```diff
@@ -678,3 +676,60 @@ def test_firefox_welcome_page1(self, render_mock):
         req.locale = "en-US"
         views.firefox_welcome_page1(req)
         render_mock.assert_called_once_with(req, "firefox/welcome/page1.html", ANY, ftl_files="firefox/welcome/page1")
+
+
+class TestSMSSendToDevice(TestCase):
+    def setUp(self):
+        self.url = "/en-US" + reverse("firefox.sms-send-to-device-post")
+        patcher = patch("bedrock.firefox.views.TWILIO_CLIENT")
+        self.mock_create = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def _do_post(self, params):
+        return self.client.post(
+            self.url,
+            data=params,
+            follow=True,
+            HTTP_X_REQUESTED_WITH="XMLHttpRequest",
+            HTTP_ACCEPT="application/json",
+        )
+
+    def test_no_data(self):
+        resp = self._do_post({})
+        assert resp.status_code == 200
+        data = resp.json()
+        assert data["success"] is False
+        assert data["errors"] == ["Invalid phone number"]
+
+    def test_invalid_number(self):
+        resp = self._do_post({"phone_number": "abcdef"})
+        assert resp.status_code == 200
+        data = resp.json()
+        assert data["success"] is False
+        assert data["errors"] == ["Invalid phone number"]
+
+    def test_invalid_number_for_region(self):
+        resp = self._do_post({"phone_number": "98 555 1212"})
+        assert resp.status_code == 200
+        data = resp.json()
+        assert data["success"] is False
+        assert data["errors"] == ["Invalid phone number for region: US"]
+
+    def test_valid_number(self):
+        resp = self._do_post({"phone_number": "415 555 1212"})
+        assert resp.status_code == 200
+        data = resp.json()
+        assert data["success"] is True
+        self.mock_create.messages.create.assert_called_once_with(
+            to="+14155551212",
+            body="test",
+            messaging_service_sid=settings.TWILIO_MESSAGING_SERVICE_SID,
+        )
+
+    def test_twilio_exception(self):
+        self.mock_create.messages.create.side_effect = TwilioRestException(status=500, uri="/", msg="Some error")
+        resp = self._do_post({"phone_number": "415 555 1212"})
+        assert resp.status_code == 200
```

Comment:
should the API respond with a code in the 400s when there's an error?

Reply:
Agreed - if we're mocking out the Twilio client, would still be nice to get a 400 as per https://www.twilio.com/docs/verify/api/v1/error-codes

Reply:
I thought so too but the email to device endpoint doesn't so I thought perhaps it was a front-end preference to send 200s but with a `success` boolean for reporting to the end user.